Add magic method `__add__`

CunliangGeng opened the pull request:
Add magic method `__add__` to `StrainCollection` class to enable the feature like `sc_new = sc1 + sc2`.


ℹ️  Tests have been checked locally. You can ignore the failed tests in github action.

CunliangGeng committed:
add `__add__` method to `StrainCollection` class

- add the `__add__` magic method
-  add unit tests for this magic method

CunliangGeng committed:
format `strain_collection.py`

gcroci2 commented on src/nplinker/strain_collection.py:
```diff
@@ -37,6 +37,16 @@ def __eq__(self, other) -> bool:
                     and self._strain_dict_name == other._strain_dict_name)
         return NotImplemented
 
+    def __add__(self, other) -> 'StrainCollection':
```
Why is the type hint a string?

CunliangGeng replied:
This is an old way to type hint the class itself (see [here](https://stackoverflow.com/questions/33533148/how-do-i-type-hint-a-method-with-the-type-of-the-enclosing-class)). I'll update it using the new way `from __future__ import annotations`.

CunliangGeng committed:
update type hints for class itself

When using `from __future__ import annotations`, the class name can be directly used as type hint; Otherwise, use string like "ClassName" as type hint.
